mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -727,6 +882,76 @@ def set_tag(self, run_id, tag):
             self._check_run_is_active(run)
             session.merge(SqlTag(run_uuid=run_id, key=tag.key, value=tag.value))
 
+    def _set_tags(self, run_id, tags, attempt=None):
+        """
+        Set multiple tags on a run
+
+        :param run_id: String ID of the run
+        :param tags: List of RunTag instances to log
+        """
+        if not tags:
+            return
+
+        valid_tags = []
+        for tag in tags:
+            _validate_tag(tag.key, tag.value)
+            valid_tags.append(tag)
+        tags = valid_tags
```

Comment:
Nit: I don't think we need to create a new list

```suggestion
        for tag in tags:
            _validate_tag(tag.key, tag.value)
```

Reply:
Yeah, absolutely better one.